src/Container/Aggregate/AggregateRepositoryFactory.php

```diff
@@ -81,7 +81,13 @@ public function __invoke(ContainerInterface $container): AggregateRepository
             throw ConfigurationException::configurationError(sprintf('Repository class %s must be a sub class of %s', $repositoryClass, AggregateRepository::class));
         }
 
-        $eventStore = $container->get(EventStore::class);
+        $eventStoreClass = $config['event_store_class'] ?? EventStore::class;
+
+        $eventStore = $container->get($eventStoreClass);
```

Comment:
change to: `$eventStore = $container->get($config['event_store']);`

Reply:
Also, implement `ProvidesDefaultConfig` and add `EventStore::class` as default option to `event_store`.